codex-rs/tui/src/onboarding/onboarding_screen.rs

```diff
@@ -53,22 +54,29 @@ pub(crate) struct OnboardingScreenArgs {
     pub chat_widget_args: ChatWidgetArgs,
     pub codex_home: PathBuf,
     pub cwd: PathBuf,
-    pub show_login_screen: bool,
     pub show_trust_screen: bool,
+    pub login_status: LoginStatus,
 }
 
 impl OnboardingScreen {
     pub(crate) fn new(args: OnboardingScreenArgs) -> Self {
+        let show_login_screen = matches!(args.login_status, LoginStatus::None)
```

Comment:
If you use a `match` does Clippy pressure you to use `matches!()` instead? I generally prefer `match` so that if a new enum variant is introduced, you are forced to deal with it.